File: iOS学习笔记/NSExceptionDemo/NSExceptionDemo/ViewController.m
//
//  ViewController.m
//  NSExceptionDemo
//
//  Created by tenny on 06/11/2017.
//  Copyright © 2017 tenny. All rights reserved.
//

#import "ViewController.h"
#import "TestException.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    TestException *t = [[TestException alloc] init];
    
    for (int i = 0; i < 10; ++i) {
        [t add];
        [t print];
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: iOS学习笔记/MemoryGraphDemo/MemoryGraphDemo/MemoryGraphDemo/ViewController.swift
//
//  ViewController.swift
//  MemoryGraphDemo
//
//  Created by tennyxu on 20/09/2017.
//  Copyright © 2017 tennyxu. All rights reserved.
//

import UIKit


class ViewController: UIViewController {
  
  @IBOutlet weak var label:UILabel!
  var ttView:TTView?
  

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    ttView = TTView(action:testf)
//    ttView = TTView(){ [unowned self] in
//      self.label.text = "hello tt";
//    }
  }
  
  func testf() {
    self.label.text = "hello tt"
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
  

  


}

